plugins/producer_plugin/include/eosio/producer_plugin/producer_plugin.hpp

```diff
@@ -82,6 +82,26 @@ class producer_plugin : public appbase::plugin<producer_plugin> {
       std::string          snapshot_name;
    };
 
+   struct snapshot_request_information {
+      uint32_t             block_spacing = 0;
+      uint32_t             start_block_num = 0;
+      uint32_t             end_block_num = 0;
+      std::string          snapshot_description = "";
+
+   };
+
+   struct snapshot_request_id_information {
+      uint32_t snapshot_request_id = 0;
+   };
+
+   struct snapshot_schedule_information : public snapshot_request_id_information, public snapshot_request_information {
+      std::optional<std::vector<snapshot_information>> pending_snapshots;
```

Comment:
I don't see the benefit of making this a `std::optional`. Isn't just an empty vector the same?

Also, Just wondering, is this pending_snapshots vector, and its management in the functions  add_pending_snapshot_info and execute_snapshot used only in the tests?

Reply:
no, this is exposed as api call, snapshot can take significant time to finalize so if there are any pending snapshot(s) triggered by a snapshot request, it will be returned in reply as json showing that it's in progress...